webapp/src/dogma/features/history/HistoryList.tsx

```diff
@@ -0,0 +1,65 @@
+import { ColumnDef, createColumnHelper } from '@tanstack/react-table';
+import { HistoryDto } from 'dogma/features/history/HistoryDto';
+import { formatDistance } from 'date-fns';
+import { DynamicDataTable } from 'dogma/common/components/table/DynamicDataTable';
+import { ViewIcon } from '@chakra-ui/icons';
+import { Button } from '@chakra-ui/react';
+import NextLink from 'next/link';
+import { ChakraLink } from 'dogma/common/components/ChakraLink';
+
+export type HistoryListProps<Data extends object> = {
+  data: Data[];
+  projectName: string;
+  repoName: string;
+  handleTabChange: Function;
+};
+
+const HistoryList = <Data extends object>({
+  data,
+  projectName,
+  repoName,
+  handleTabChange,
+}: HistoryListProps<Data>) => {
+  const columnHelper = createColumnHelper<HistoryDto>();
+  const columns = [
+    columnHelper.accessor((row: HistoryDto) => parseInt(row.revision.revisionNumber), {
+      cell: (info) => (
+        <ChakraLink
+          fontWeight="semibold"
+          href={`/app/projects/${projectName}/repos/${repoName}/list/${info.getValue()}/`}
+          onClick={() => handleTabChange(0)}
+        >
+          {info.getValue()}
+        </ChakraLink>
+      ),
+      header: 'Revision',
+    }),
+    columnHelper.accessor((row: HistoryDto) => row.summary, {
+      cell: (info) => info.getValue(),
+      header: 'Summary',
+    }),
+
+    columnHelper.accessor((row: HistoryDto) => row.author.name, {
+      cell: (info) => info.getValue(),
+      header: 'Author',
+    }),
+    columnHelper.accessor((row: HistoryDto) => row.timestamp, {
+      cell: (info) => formatDistance(new Date(info.getValue()), new Date(), { addSuffix: true }),
```

Comment:
It would be useful to show ISO-8601 format with a tool tip or somehow.
Some users may want to know the exact time when a file is updated.

Reply:
Updated to the following.

<img width="1266" alt="Screen Shot 2022-12-14 at 10 45 57" src="https://user-images.githubusercontent.com/5079588/207500526-529ecc95-a217-4bde-bc03-45e88ae8e4b1.png">